### PR TITLE
Add data model for a Message in the engagement database

### PR DESCRIPTION
**Pull request by as2388**

### CODE REVIEW

File: engagement_database/data_models.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+from core_data_modules.data_models import Label
+
+
+class MessageStatuses(object):
+    LIVE = "live"              # Message is part of the 'live' data, and should be used in analysis.
+    STALE = "stale"            # Message is stale and we should attempt to get a newer answer if we can. This message
+                               # can be used until we get a newer answer.
+    ARCHIVED = "archived"      # This message is no longer relevant and should be ignored by analysis.
+    TOMBSTONED = "tombstoned"  # Message is tombstoned, so personally identifiable details have been removed.
```
Comment (DanielMwendwa) [inline engagement_database/data_models.py:11]: I suggest using `NON_PII = "Non-PII"` for Non-personally identifiable information.

**Review by DanielMwendwa — APPROVED** [review]

File: engagement_database/data_models.py (hunk 1 above)
Comment (as2388) [reply engagement_database/data_models.py:11]: How about "anonymised"?

File: engagement_database/data_models.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        self.channel_operator = channel_operator
+        self.status = status
+        self.dataset = dataset
+        self.labels = labels
+        self.message_id = message_id
+        self.coda_id = coda_id
+        self.last_updated = last_updated
+
+    def to_dict(self):
+        d = {
```
Comment (IsaackMwenda) [inline engagement_database/data_models.py:71]:
```suggestion
        message = {
```

**Review by IsaackMwenda — APPROVED** [review]
LGTM - thanks

File: engagement_database/data_models.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        }
+
+        if self.coda_id is not None:
+            d["coda_id"] = self.coda_id
+
+        return d
+
+    @classmethod
+    def from_dict(cls, d):
+        return cls(
```
[the remark this answers is not in the document]
Comment (as2388) [reply engagement_database/data_models.py:91]: For sure in the history sub-collection. I don't think we need to duplicate here, but happy to discuss offline/add later.

**Pull request merged**